components/alert/AlertMap.vue

```diff
@@ -64,43 +217,744 @@ export default Vue.extend({
+        }
+      });
+      indicatorText.setIcon(text).addTo(this.textBadges);
+      return badge;
+    },
+    warpgate(faction: Faction): MapRegion | undefined {
+      var zone = this.alert.zone;
+      if(!zone){
+        console.error("Unable to find warpgate of undefined zone!");
+        return undefined;
+      }
+      var warpgates: number[] | undefined = zoneToWarpgateArray.get(zone);
+      if(warpgates == undefined){
+        return undefined;
+      }
+      var region: MapRegion | undefined = undefined;
+      warpgates.forEach((facilityId: number) => {
+        if(region) {
+          return;
+        }
+        if(this.mapRegions.get(facilityId)?.faction === faction){
+          region = this.mapRegions.get(facilityId);
+        }
+      });
+      return region;
+    },
+    cutoff(facility: MapRegion): boolean {
+      // This could use improvement, probably should be moved to MapRegion as well (Need to record all links in MapRegion first)
+      // Currently this performs a depth first search of all connected territories starting at the faction's warpgate
+      //    and returns true if it finds *facility* during the search.
+      if(facility.facilityType == FacilityType.WARPGATE){
+        return false;
+      }
+      // This array is being treated as a stack
+      var frontier = [this.warpgate(facility.faction)];
+      // Easy improvement here - don't use an array for the visited data structure
+      var visited: MapRegion[] = [];
+      while(frontier.length !== 0){
+        var curr = frontier.pop();
+        if(!curr){
+          return false;
+        }
+        if(curr.id === facility.id){
+          return false;
+        }
+        // Linkstamps is a record of the shape { "1234 2345": <leaflet Layer id>, ... }, 
+        //    the keys are two facility ids concatenated with a space. Not every permutation is present
+        //    since most (but not all of course) facilities in the census do not reciprocate their map links
+        //    (AKA this describes something that is close to (but is not) a directed graph)
+        Object.keys(this.linkStamps).forEach((pair) => {
+          if(pair.includes("bg")){
+            return;
+          }
+          var facilities = pair.split(" ").map((val) => parseInt(val));
+          if(!(facilities[0] === curr?.id || facilities[1] === curr?.id))
+            return;
+          var connection = facilities[0] === curr?.id ? this.mapRegions.get(facilities[1]) : this.mapRegions.get(facilities[0]);
+          var wasVisited = visited.find((region) => { return region.id === connection?.id; });
+          if (wasVisited){
+            return;
+          }
+          if(connection?.faction === facility.faction){
+            frontier.push(connection);
+          }
+        });
+        visited.push(curr);
+      }
+      return true;
```

Comment:
I've reworked this to be much less convoluted by ensuring that all regions have a list of all the connecting regions they are linked to, and using that to figure out which connections to explore rather than a weird string pair system like I was using.

Reply:
Yep, excellent refactor!